spring-cloud-stream-schema/src/main/java/org/springframework/cloud/stream/schema/client/CachingRegistryClient.java

```diff
@@ -47,8 +47,8 @@ public CachingRegistryClient(SchemaRegistryClient delegate) {
 	@Override
 	public SchemaRegistrationResponse register(String subject, String format, String schema) {
 		SchemaRegistrationResponse response = delegate.register(subject, format, schema);
-		cacheManager.getCache(ID_CACHE).put(response.getSchemaReference(), schema);
-		cacheManager.getCache(REF_CACHE).put(response.getId(), schema);
+		cacheManager.getCache(ID_CACHE).put(response.getId(), schema);
+		cacheManager.getCache(REF_CACHE).put(response.getSchemaReference(), schema);
```

Comment:
ouch! Good catch!

Reply:
Can you guys back port this?

Reply:
Sure, we can back port this, but I now wonder, why do we need two caches? We use the same delegate, cache is just a K/V pair. Should it really just be `schemaCache`?
```
cacheManager.getCache("org.springframework.cloud.stream.schema.client.schemaCache").put(response.getId(), schema);
cacheManager.getCache("org.springframework.cloud.stream.schema.client.schemaCache").put(response.getSchemaReference(), schema);
```
@viniciusccarvalho ^^

Reply:
We decided that since they are two different entities we would put them on different caches as it's good practice with caches usually. One holds ints->schema, the other object->schema